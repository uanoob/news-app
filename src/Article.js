import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Article extends Component {
  state = {
    isOpen: false,
  };

  toggleOpen = () => {
    const { isOpen } = this.state;
    this.setState({
      isOpen: !isOpen,
    });
  };

  render() {
    const { article } = this.props;
    const { isOpen } = this.state;
    return (
      <div>
        <h3>{article.title}</h3>
        <button type="button" onClick={this.toggleOpen}>
          {isOpen ? 'Close' : 'Open'}
        </button>
        {isOpen ? <section>{article.text}</section> : null}
      </div>
    );
  }
}

Article.propTypes = {
  article: PropTypes.shape({
    comments: PropTypes.arrayOf(PropTypes.string),
    date: PropTypes.string.isRequired,
    text: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    _id: PropTypes.string.isRequired,
  }).isRequired,
};

export default Article;
