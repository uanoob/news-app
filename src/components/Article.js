import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { withStyles } from '@material-ui/core/styles';
import Collapse from '@material-ui/core/Collapse';
import Card from '@material-ui/core/Card';
import CardHeader from '@material-ui/core/CardHeader';
import CardContent from '@material-ui/core/CardContent';
import CardActions from '@material-ui/core/CardActions';
import Avatar from '@material-ui/core/Avatar';
import IconButton from '@material-ui/core/IconButton';
import Typography from '@material-ui/core/Typography';
import red from '@material-ui/core/colors/red';
import FavoriteIcon from '@material-ui/icons/Favorite';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
import ForwardIcon from '@material-ui/icons/ForwardOutlined';
import CommentIcon from '@material-ui/icons/Comment';

import Comments from './Comments';
import CommentForm from './CommentForm';
import handleAuthorAvatar from '../utils/avatar.utils';
import stringToColor from '../utils/stringToColor.utils';

const styles = theme => ({
  card: {
    width: '100%',
    backgroundColor: theme.palette.background.paper,
  },
  actions: {
    display: 'flex',
  },
  expand: {
    transform: 'rotate(0deg)',
    transition: theme.transitions.create('transform', {
      duration: theme.transitions.duration.shortest,
    }),
    marginLeft: 'auto',
    [theme.breakpoints.up('sm')]: {
      marginRight: -8,
    },
  },
  expandOpen: {
    transform: 'rotate(180deg)',
  },
  avatar: {
    backgroundColor: red[500],
  },
});

class Article extends Component {
  state = {
    expanded: false,
    dialog: false,
  };

  handleExpandClick = () => {
    this.setState(state => ({ expanded: !state.expanded }));
  };

  handleDialogClick = () => {
    const { dialog } = this.state;
    this.setState({
      dialog: !dialog,
    });
  };

  render() {
    const { classes, article } = this.props;
    const { expanded, dialog } = this.state;
    return (
      <Card className={classes.card}>
        <CardHeader
          avatar={(
            <Avatar
              aria-label="Recipe"
              style={{
                backgroundColor: stringToColor(
                  article.author_name || 'Anonymous',
                ),
              }}
            >
              {handleAuthorAvatar(article.author_name || 'Anonymous')}
            </Avatar>
)}
          action={(
            <IconButton>
              <ForwardIcon />
            </IconButton>
)}
          title={article.title}
          subheader={article.posted_at}
        />

        <CardContent>
          <Typography component="p">{article.text}</Typography>
        </CardContent>

        <CardActions className={classes.actions} disableActionSpacing>
          <IconButton aria-label="Add to favorites">
            <FavoriteIcon />
          </IconButton>
          <IconButton aria-label="Message" onClick={this.handleDialogClick}>
            <CommentIcon />
          </IconButton>

          <CommentForm
            articleId={article._id}
            dialog={dialog}
            handleDialogClick={this.handleDialogClick}
            title="Add a comment"
            description="Please enter your comment here."
          />

          <IconButton
            className={classnames(classes.expand, {
              [classes.expandOpen]: expanded,
            })}
            onClick={this.handleExpandClick}
            aria-expanded={expanded}
            aria-label="Show more"
          >
            <ExpandMoreIcon />
          </IconButton>
        </CardActions>

        <Collapse in={expanded} timeout="auto" unmountOnExit>
          <CardContent>
            <Comments articleId={article._id} />
          </CardContent>
        </Collapse>
      </Card>
    );
  }
}

Article.propTypes = {
  classes: PropTypes.shape({
    card: PropTypes.string.isRequired,
    actions: PropTypes.string.isRequired,
    expand: PropTypes.string.isRequired,
    expandOpen: PropTypes.string.isRequired,
    avatar: PropTypes.string.isRequired,
  }).isRequired,
  article: PropTypes.shape({
    _id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    text: PropTypes.string.isRequired,
    author_id: PropTypes.string.isRequired,
    author_name: PropTypes.string.isRequired,
    posted_at: PropTypes.string.isRequired,
  }).isRequired,
};

export default withStyles(styles)(Article);
