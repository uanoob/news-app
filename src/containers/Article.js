import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Comments from '../components/Comments';
import { getCommentById } from '../api';

class Article extends Component {
  static propTypes = {
    article: PropTypes.shape({
      comments: PropTypes.arrayOf(PropTypes.string),
      date: PropTypes.string.isRequired,
      text: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired,
      _id: PropTypes.string.isRequired,
    }).isRequired,
  };

  state = {
    isOpen: false,
    comments: [],
  };

  componentDidMount() {
    this.getComments();
  }

  getComments = () => {
    const { article } = this.props;
    const arr = [];
    if (article.comments.length !== 0) {
      article.comments.map(async (id) => {
        const comment = await getCommentById(id);
        arr.push(comment);
      });
    }
    return this.setState({
      comments: arr,
    });
  };

  toggleOpen = () => {
    const { isOpen } = this.state;
    this.setState({
      isOpen: !isOpen,
    });
  };

  render() {
    const { article } = this.props;
    const { isOpen, comments } = this.state;
    return (
      <div>
        <h3>{article.title}</h3>
        <button type="button" onClick={this.toggleOpen}>
          {isOpen ? 'Close' : 'Open'}
        </button>
        {isOpen ? (
          <section>
            <p>{article.text}</p>
            <Comments comments={comments} />
          </section>
        ) : null}
      </div>
    );
  }
}

export default Article;
