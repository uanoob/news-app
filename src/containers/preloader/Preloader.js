import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import CircularProgress from '@material-ui/core/CircularProgress';

const styles = () => ({
  progress: {
    marginTop: 50,
  },
});

function CircularIndeterminate(props) {
  const { classes } = props;
  return (
    <div>
      <CircularProgress
        className={classes.progress}
        color="primary"
        size={100}
      />
    </div>
  );
}

CircularIndeterminate.propTypes = {
  classes: PropTypes.shape({
    progress: PropTypes.string.isRequired,
  }).isRequired,
};

export default withStyles(styles)(CircularIndeterminate);
