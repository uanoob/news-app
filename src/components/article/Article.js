import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { withStyles } from '@material-ui/core/styles';
import Collapse from '@material-ui/core/Collapse';
import Card from '@material-ui/core/Card';
import CardHeader from '@material-ui/core/CardHeader';
import CardContent from '@material-ui/core/CardContent';
import CardActions from '@material-ui/core/CardActions';
import Avatar from '@material-ui/core/Avatar';
import IconButton from '@material-ui/core/IconButton';
import Typography from '@material-ui/core/Typography';
import red from '@material-ui/core/colors/red';
import FavoriteIcon from '@material-ui/icons/Favorite';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
import ForwardIcon from '@material-ui/icons/ForwardOutlined';
import CommentIcon from '@material-ui/icons/Comment';
import DeleteIcon from '@material-ui/icons/Delete';
import EditIcon from '@material-ui/icons/Edit';
import { deleteArticle, clearArticle } from '../../store/actions';
import Comments from '../comment/Comments';
import CommentForm from '../comment/CommentForm';
import ArticleForm from './ArticleForm';
import Preloader from '../../containers/preloader/Preloader';
import {
  handleAuthorAvatar,
  stringToColor,
  cutStringLength,
} from '../../utils';

const styles = theme => ({
  card: {
    width: '100%',
    backgroundColor: theme.palette.background.paper,
  },
  loader: {
    textAlign: 'center',
  },
  actions: {
    display: 'flex',
  },
  expand: {
    transform: 'rotate(0deg)',
    transition: theme.transitions.create('transform', {
      duration: theme.transitions.duration.shortest,
    }),
    marginLeft: 'auto',
    [theme.breakpoints.up('sm')]: {
      marginRight: -8,
    },
  },
  expandOpen: {
    transform: 'rotate(180deg)',
  },
  avatar: {
    backgroundColor: red[500],
  },
});

export class ArticleComponent extends Component {
  state = {
    expanded: false,
    articleDialog: false,
    commentDialog: false,
  };

  componentWillUnmount() {
    this.handleClearArticle();
  }

  handleClearArticle = () => {
    const { onClearArticle, currentArticleId } = this.props;
    return currentArticleId ? onClearArticle() : null;
  };

  handleExpandClick = () => {
    this.setState(state => ({ expanded: !state.expanded }));
  };

  handleArticleClick = () => {
    const { articleDialog } = this.state;
    this.setState({
      articleDialog: !articleDialog,
    });
  };

  handleCommentClick = () => {
    const { commentDialog } = this.state;
    this.setState({
      commentDialog: !commentDialog,
    });
  };

  handleCurrentArticle = (articleId) => {
    const { history } = this.props;
    history.push(`/article/${articleId}`);
  };

  handleDeleteArticle = (articleId) => {
    const { onDeleteArticle, userId } = this.props;
    onDeleteArticle(articleId, userId);
    const { history } = this.props;
    history.push(`/author/${userId}`);
  };

  handleIsAuthor = () => {
    const { article, userId } = this.props;
    return article.author_id === userId ? (
      <div>
        <IconButton
          id="article-edit-button"
          type="button"
          onClick={this.handleArticleClick}
        >
          <EditIcon />
        </IconButton>
        <IconButton
          id="article-delete-button"
          type="button"
          onClick={() => this.handleDeleteArticle(article._id)}
        >
          <DeleteIcon />
        </IconButton>
      </div>
    ) : null;
  };

  render() {
    const {
      classes, article, userId, isLoading,
    } = this.props;
    const { expanded, articleDialog, commentDialog } = this.state;
    return isLoading ? (
      <div className={classes.loader}>
        <Preloader />
      </div>
    ) : (
      <Card className={classes.card}>
        <CardHeader
          avatar={(
            <Avatar
              aria-label="Recipe"
              style={{
                backgroundColor: stringToColor(article.author_name),
              }}
            >
              {handleAuthorAvatar(article.author_name)}
            </Avatar>
)}
          action={
            !userId ? (
              <IconButton
                onClick={() => this.handleCurrentArticle(article._id)}
              >
                <ForwardIcon />
              </IconButton>
            ) : (
              this.handleIsAuthor()
            )
          }
          title={article.title}
          subheader={article.posted_at}
        />

        <CardContent>
          <Typography component="p">
            {userId ? article.text : cutStringLength(article.text)}
          </Typography>
        </CardContent>

        <CardActions className={classes.actions} disableActionSpacing>
          <IconButton aria-label="Add to favorites">
            <FavoriteIcon />
          </IconButton>
          <IconButton aria-label="Message" onClick={this.handleCommentClick}>
            <CommentIcon />
          </IconButton>

          <CommentForm
            articleId={article._id}
            dialog={commentDialog}
            handleDialogClick={this.handleCommentClick}
            title="Add a comment"
            description="Please enter your comment here."
          />

          <ArticleForm
            articleId={article._id}
            articleTitle={article.title}
            articleText={article.text}
            dialog={articleDialog}
            handleDialogClick={this.handleArticleClick}
            header="Edit an article"
            description="Please edit your article here."
          />

          <IconButton
            className={classnames(classes.expand, {
              [classes.expandOpen]: expanded,
            })}
            onClick={this.handleExpandClick}
            aria-expanded={expanded}
            aria-label="Show more"
          >
            <ExpandMoreIcon />
          </IconButton>
        </CardActions>

        <Collapse in={expanded} timeout="auto" unmountOnExit>
          <CardContent>
            <Comments articleId={article._id} />
          </CardContent>
        </Collapse>
      </Card>
    );
  }
}

ArticleComponent.defaultProps = {
  userId: '',
  currentArticleId: '',
};

ArticleComponent.propTypes = {
  classes: PropTypes.shape({
    card: PropTypes.string.isRequired,
    actions: PropTypes.string.isRequired,
    expand: PropTypes.string.isRequired,
    expandOpen: PropTypes.string.isRequired,
    avatar: PropTypes.string.isRequired,
  }).isRequired,
  article: PropTypes.shape({
    _id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    text: PropTypes.string.isRequired,
    author_id: PropTypes.string.isRequired,
    author_name: PropTypes.string.isRequired,
    created_at: PropTypes.string.isRequired,
    updated_at: PropTypes.string.isRequired,
  }).isRequired,
  history: PropTypes.shape({ push: PropTypes.func.isRequired }).isRequired,
  userId: PropTypes.string,
  onDeleteArticle: PropTypes.func.isRequired,
  onClearArticle: PropTypes.func.isRequired,
  currentArticleId: PropTypes.string,
  isLoading: PropTypes.bool.isRequired,
};

const mapStateToProps = state => ({
  currentArticleId: state.article.article._id,
  isLoading: state.article.loading,
});

const mapDispatchToProps = {
  onDeleteArticle: deleteArticle,
  onClearArticle: clearArticle,
};

const Article = withRouter(
  connect(
    mapStateToProps,
    mapDispatchToProps,
  )(withStyles(styles)(ArticleComponent)),
);

export default Article;
