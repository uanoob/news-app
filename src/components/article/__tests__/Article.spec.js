import React from 'react';
import { shallow } from 'enzyme';
import { ArticleComponent } from '../Article';
import Preloader from '../../../containers/preloader/Preloader';

describe('Article', () => {
  // const initialState = {
  //   expanded: false,
  //   articleDialog: false,
  //   commentDialog: false,
  // };
  const mockDeleteArticle = jest.fn();
  const mockClearArticle = jest.fn();
  const props = {
    classes: {
      card: '',
      actions: '',
      expand: '',
      expandOpen: '',
      avatar: '',
    },
    article: {
      _id: '',
      title: '',
      text: '',
      author_id: 'test_author_id',
      author_name: '',
      created_at: '',
      updated_at: '',
    },
    history: { push: () => {} },
    userId: '',
    onDeleteArticle: mockDeleteArticle,
    onClearArticle: mockClearArticle,
    currentArticleId: '',
    isLoading: false,
  };

  describe('ArticleComponent initial', () => {
    const wrapper = shallow(<ArticleComponent {...props} />);
    it('render correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
    it('not render <Preloader />', () => {
      expect(wrapper.find(Preloader)).toHaveLength(0);
    });
  });
  describe('ArticleComponent isLoading', () => {
    const nextProps = {
      ...props,
      isLoading: true,
    };
    const wrapper = shallow(<ArticleComponent {...nextProps} />);
    it('render <Preloader />', () => {
      expect(wrapper.find(Preloader)).toHaveLength(1);
    });
  });
  describe('ArticleComponent componentWillUnmount', () => {
    const wrapper = shallow(<ArticleComponent {...props} />);
    const instance = wrapper.instance();
    jest.spyOn(instance, 'handleClearArticle');
    wrapper.unmount();
    it('method handleClearArticle should have to be called', () => {
      expect(instance.handleClearArticle).toBeCalledTimes(1);
    });
  });
  describe('ArticleComponent & userId', () => {
    const nextProps = {
      ...props,
      userId: 'test_author_id',
    };
    const wrapper = shallow(<ArticleComponent {...nextProps} />);
    describe('when clicking the Delete button', () => {
      wrapper.find('#article-delete-button').simulate('click');
      it('method onDeleteArticle should been called', () => {
        expect(mockDeleteArticle).toBeCalledTimes(1);
      });
    });
  });
});
