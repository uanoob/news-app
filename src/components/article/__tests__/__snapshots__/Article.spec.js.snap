// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article ArticleComponent initial render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArticleComponent
    article={
      Object {
        "_id": "",
        "author_id": "test_author_id",
        "author_name": "",
        "created_at": "",
        "text": "",
        "title": "",
        "updated_at": "",
      }
    }
    classes={
      Object {
        "actions": "",
        "avatar": "",
        "card": "",
        "expand": "",
        "expandOpen": "",
      }
    }
    currentArticleId=""
    history={
      Object {
        "push": [Function],
      }
    }
    isLoading={false}
    onClearArticle={[MockFunction]}
    onDeleteArticle={[MockFunction]}
    userId=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <WithStyles(CardHeader)
          action={
            <WithStyles(IconButton)
              onClick={[Function]}
            >
              <pure(ForwardOutlinedIcon) />
            </WithStyles(IconButton)>
          }
          avatar={
            <WithStyles(Avatar)
              aria-label="Recipe"
              style={
                Object {
                  "backgroundColor": "#000000",
                }
              }
            >
              
            </WithStyles(Avatar)>
          }
          title=""
        />,
        <WithStyles(CardContent)>
          <WithStyles(Typography)
            component="p"
          >
            
          </WithStyles(Typography)>
        </WithStyles(CardContent)>,
        <WithStyles(CardActions)
          className=""
          disableActionSpacing={true}
        >
          <WithStyles(IconButton)
            aria-label="Add to favorites"
          >
            <pure(FavoriteIcon) />
          </WithStyles(IconButton)>
          <WithStyles(IconButton)
            aria-label="Message"
            onClick={[Function]}
          >
            <pure(CommentIcon) />
          </WithStyles(IconButton)>
          <Connect(WithStyles(CommentFormComponent))
            articleId=""
            description="Please enter your comment here."
            dialog={false}
            handleDialogClick={[Function]}
            title="Add a comment"
          />
          <Connect(WithStyles(ArticleFormComponent))
            articleId=""
            articleText=""
            articleTitle=""
            description="Please edit your article here."
            dialog={false}
            handleDialogClick={[Function]}
            header="Edit an article"
          />
          <WithStyles(IconButton)
            aria-expanded={false}
            aria-label="Show more"
            className=""
            onClick={[Function]}
          >
            <pure(ExpandMoreIcon) />
          </WithStyles(IconButton)>
        </WithStyles(CardActions)>,
        <WithStyles(Collapse)
          in={false}
          timeout="auto"
          unmountOnExit={true}
        >
          <WithStyles(CardContent)>
            <Connect(WithStyles(CommentsComponent))
              articleId=""
            />
          </WithStyles(CardContent)>
        </WithStyles(Collapse)>,
      ],
      "className": "",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "action": <WithStyles(IconButton)
            onClick={[Function]}
          >
            <pure(ForwardOutlinedIcon) />
          </WithStyles(IconButton)>,
          "avatar": <WithStyles(Avatar)
            aria-label="Recipe"
            style={
              Object {
                "backgroundColor": "#000000",
              }
            }
          >
            
          </WithStyles(Avatar)>,
          "subheader": undefined,
          "title": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithStyles(Typography)
            component="p"
          >
            
          </WithStyles(Typography)>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "",
            "component": "p",
          },
          "ref": null,
          "rendered": "",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(IconButton)
              aria-label="Add to favorites"
            >
              <pure(FavoriteIcon) />
            </WithStyles(IconButton)>,
            <WithStyles(IconButton)
              aria-label="Message"
              onClick={[Function]}
            >
              <pure(CommentIcon) />
            </WithStyles(IconButton)>,
            <Connect(WithStyles(CommentFormComponent))
              articleId=""
              description="Please enter your comment here."
              dialog={false}
              handleDialogClick={[Function]}
              title="Add a comment"
            />,
            <Connect(WithStyles(ArticleFormComponent))
              articleId=""
              articleText=""
              articleTitle=""
              description="Please edit your article here."
              dialog={false}
              handleDialogClick={[Function]}
              header="Edit an article"
            />,
            <WithStyles(IconButton)
              aria-expanded={false}
              aria-label="Show more"
              className=""
              onClick={[Function]}
            >
              <pure(ExpandMoreIcon) />
            </WithStyles(IconButton)>,
          ],
          "className": "",
          "disableActionSpacing": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "aria-label": "Add to favorites",
              "children": <pure(FavoriteIcon) />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "aria-label": "Message",
              "children": <pure(CommentIcon) />,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "articleId": "",
              "description": "Please enter your comment here.",
              "dialog": false,
              "handleDialogClick": [Function],
              "title": "Add a comment",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "articleId": "",
              "articleText": "",
              "articleTitle": "",
              "description": "Please edit your article here.",
              "dialog": false,
              "handleDialogClick": [Function],
              "header": "Edit an article",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "aria-expanded": false,
              "aria-label": "Show more",
              "children": <pure(ExpandMoreIcon) />,
              "className": "",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithStyles(CardContent)>
            <Connect(WithStyles(CommentsComponent))
              articleId=""
            />
          </WithStyles(CardContent)>,
          "in": false,
          "timeout": "auto",
          "unmountOnExit": true,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Connect(WithStyles(CommentsComponent))
              articleId=""
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "articleId": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithStyles(CardHeader)
            action={
              <WithStyles(IconButton)
                onClick={[Function]}
              >
                <pure(ForwardOutlinedIcon) />
              </WithStyles(IconButton)>
            }
            avatar={
              <WithStyles(Avatar)
                aria-label="Recipe"
                style={
                  Object {
                    "backgroundColor": "#000000",
                  }
                }
              >
                
              </WithStyles(Avatar)>
            }
            title=""
          />,
          <WithStyles(CardContent)>
            <WithStyles(Typography)
              component="p"
            >
              
            </WithStyles(Typography)>
          </WithStyles(CardContent)>,
          <WithStyles(CardActions)
            className=""
            disableActionSpacing={true}
          >
            <WithStyles(IconButton)
              aria-label="Add to favorites"
            >
              <pure(FavoriteIcon) />
            </WithStyles(IconButton)>
            <WithStyles(IconButton)
              aria-label="Message"
              onClick={[Function]}
            >
              <pure(CommentIcon) />
            </WithStyles(IconButton)>
            <Connect(WithStyles(CommentFormComponent))
              articleId=""
              description="Please enter your comment here."
              dialog={false}
              handleDialogClick={[Function]}
              title="Add a comment"
            />
            <Connect(WithStyles(ArticleFormComponent))
              articleId=""
              articleText=""
              articleTitle=""
              description="Please edit your article here."
              dialog={false}
              handleDialogClick={[Function]}
              header="Edit an article"
            />
            <WithStyles(IconButton)
              aria-expanded={false}
              aria-label="Show more"
              className=""
              onClick={[Function]}
            >
              <pure(ExpandMoreIcon) />
            </WithStyles(IconButton)>
          </WithStyles(CardActions)>,
          <WithStyles(Collapse)
            in={false}
            timeout="auto"
            unmountOnExit={true}
          >
            <WithStyles(CardContent)>
              <Connect(WithStyles(CommentsComponent))
                articleId=""
              />
            </WithStyles(CardContent)>
          </WithStyles(Collapse)>,
        ],
        "className": "",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "action": <WithStyles(IconButton)
              onClick={[Function]}
            >
              <pure(ForwardOutlinedIcon) />
            </WithStyles(IconButton)>,
            "avatar": <WithStyles(Avatar)
              aria-label="Recipe"
              style={
                Object {
                  "backgroundColor": "#000000",
                }
              }
            >
              
            </WithStyles(Avatar)>,
            "subheader": undefined,
            "title": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(Typography)
              component="p"
            >
              
            </WithStyles(Typography)>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "",
              "component": "p",
            },
            "ref": null,
            "rendered": "",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(IconButton)
                aria-label="Add to favorites"
              >
                <pure(FavoriteIcon) />
              </WithStyles(IconButton)>,
              <WithStyles(IconButton)
                aria-label="Message"
                onClick={[Function]}
              >
                <pure(CommentIcon) />
              </WithStyles(IconButton)>,
              <Connect(WithStyles(CommentFormComponent))
                articleId=""
                description="Please enter your comment here."
                dialog={false}
                handleDialogClick={[Function]}
                title="Add a comment"
              />,
              <Connect(WithStyles(ArticleFormComponent))
                articleId=""
                articleText=""
                articleTitle=""
                description="Please edit your article here."
                dialog={false}
                handleDialogClick={[Function]}
                header="Edit an article"
              />,
              <WithStyles(IconButton)
                aria-expanded={false}
                aria-label="Show more"
                className=""
                onClick={[Function]}
              >
                <pure(ExpandMoreIcon) />
              </WithStyles(IconButton)>,
            ],
            "className": "",
            "disableActionSpacing": true,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "aria-label": "Add to favorites",
                "children": <pure(FavoriteIcon) />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "aria-label": "Message",
                "children": <pure(CommentIcon) />,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "articleId": "",
                "description": "Please enter your comment here.",
                "dialog": false,
                "handleDialogClick": [Function],
                "title": "Add a comment",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "articleId": "",
                "articleText": "",
                "articleTitle": "",
                "description": "Please edit your article here.",
                "dialog": false,
                "handleDialogClick": [Function],
                "header": "Edit an article",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "aria-expanded": false,
                "aria-label": "Show more",
                "children": <pure(ExpandMoreIcon) />,
                "className": "",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(CardContent)>
              <Connect(WithStyles(CommentsComponent))
                articleId=""
              />
            </WithStyles(CardContent)>,
            "in": false,
            "timeout": "auto",
            "unmountOnExit": true,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Connect(WithStyles(CommentsComponent))
                articleId=""
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "articleId": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
